src/librustc/mir/interpret/allocation.rs

```diff
@@ -15,16 +15,17 @@ use std::borrow::Cow;
 
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
 pub struct Allocation<Tag=(),Extra=()> {
-    /// The actual bytes of the allocation.
-    /// Note that the bytes of a pointer represent the offset of the pointer
-    pub bytes: Vec<u8>,
+    /// The byte allocation and definedness.
+    /// Bytes of fully undefined allocations are allocated lazily on potentially mutable changes.
+    pub alloc: Option<AllocationBytes>,
     /// Maps from byte addresses to extra data for each pointer.
     /// Only the first byte of a pointer is inserted into the map; i.e.,
     /// every entry in this map applies to `pointer_size` consecutive bytes starting
     /// at the given offset.
     pub relocations: Relocations<Tag>,
-    /// Denotes undefined memory. Reading from undefined memory is forbidden in miri
-    pub undef_mask: UndefMask,
+    /// The size of the allocation.
+    /// Used after initialiation of an undef region that is not allocated.
```

Comment:
I find this comment confusing. What about "If `alloc` is not `None`, this is the same as `bytes.len()`"?